### DISCUSSION TITLE
make Area of the skin editing conserve the aspect ratio of the window

### DISCUSSION THREAD

**Discussion by Avelia-tr** [ppy/osu#27804] [Ideas]
To make thing more clear I'm talking about this area highlighted in red here
![image](https://github.com/ppy/osu/assets/154550569/d845b509-14ff-4544-ae61-c61ca97c4d09)

Currently it's really unintuitive to edit a skin. I believe the relative nature of the elements is what makes it unintuitive. To fix that I just propose to make this window smaller (like the highlighted red rectangle) this would make the process wxay more straightfroward and avoid any back and forth in and out of the skin editor. 
![image](https://github.com/ppy/osu/assets/154550569/df473b2e-259d-4996-a40d-e038e972f8ab)

**Comment by Walavouchey** [comment]
this is intended behaviour. the idea is to allow skins to be designed agnostic to aspect ratio

however, i can recommend hitting alt+enter for windowed mode and changing the window size such that the centre area becomes 16:9, for example with ~1564x720

**Reply by Avelia-tr** [reply]
I know this is intended behaviour, that's why I didn't put this under the problem tag but under the idea one. 

Your recommendation is cool, but it shouldn't be like resizing the window shouldn't be something that's done. 
I believe we can keep the agnostic to aspect ratio aspect while allowing for more intuitive skinning experience by doing what I propose.

**Comment by peppy** [comment]
Can you describe which elements you're having trouble with? It sounds like you're misusing anchor / origin, because if you set these correctly it will all adjust correctly.

For example, if your issue is with the two components you have at the top-left, then you need to set them both to top left anchor at very least. This should be done for you, though, unless you've manually overridde.

**Reply by Avelia-tr** [reply]
this system of anchor is generally misunderstood. I don't think I saw anybody that don't needed an explanation or that just think of it as some dark magic. (I know this is anecdotal evidence but I don't think I can give you better thing) 

I've seen a lot of people saying that it was tiresome to get thing at the right place/ doing something precise like in this video https://youtu.be/ou2wtN2XQAo?si=A3WrCjMFhN_oskSH&t=407 can be really akward. 

example shown in the video 
this leads to 
![image](https://github.com/ppy/osu/assets/154550569/f239de40-18e5-4f72-8f22-0b16227fb930)
this  
![image](https://github.com/ppy/osu/assets/154550569/3236c9d0-cec8-4a3e-bf6c-218a4e855310)
wich is annoying because it ask the player to come back and forth. 
 
while I like the whole agnostic to aspect ratio thing I fail to really understand why it would really useful to show it like that. 
In my opinion a skin editor preview should look like how it'll look in game not how it'll look in a different aspect ratio. (I'll maintain that I have nothing against the actual system of how it works just how it's shown)

**Reply by Avelia-tr** [reply]
Nevermind the issue was explained to me in private ignore this please


oh and while doing further testing I noticed that the "nametag" has a bug when changing pseudo 
this is what my nametag look like in gameplay 
![image](https://github.com/ppy/osu/assets/154550569/9d89fa31-9c8e-4dc6-8f4d-3c25fcec0345)

and this is how the object looks in the skin editor. 
![image](https://github.com/ppy/osu/assets/154550569/6918ea34-17f9-4644-8713-4c142378c768)
but I don't know where to really report it

**Reply by Walavouchey** [reply]
@Avelia-tr you could be pointing out two issues, but neither are bugs

1. the username and avatar are different because they reflect the current player. in the skin editor, you're looking at autoplay. try opening the skin editor while viewing a replay
2. "Autoplay" overlaps the avatar because your origin is set incorrectly. it should be either `TopLeft`, `CentreLeft` or `BottomLeft`, which will make the username expand to the right

**Reply by Avelia-tr** [reply]
yeah that was explained to me in private. The origin is set by default to up center for this wich is what cause this. And like you daid this isn't a bug it's just the anchor system not being that intuitive when it's not explained

**Reply by Joehuu** [reply]
I think there should be a "Same as anchor" setting as a default, considering the codebase has origin = anchor most of the time.

**Comment by cl8n** [comment]
to give one example of where I have run into the same issue -- I have ArgonSongProgress rotated sideways so that it fills from bottom to top, but it appears wildly different size in the editor vs. in gameplay due to the change in aspect ratio. I basically just guessed and checked until I got it right, and I don't know if I can make this "agnostic to aspect ratio" somehow.

**Reply by peppy** [reply]
rotating relative elements is an edge case that has an open tracking issue somewhere. that's not what is happening here.
